File: 0053-maximum-subarray/0053-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        cuurent = 0
        mxs =nums[0]
        for i in nums:
            cuurent+=i
            if(cuurent>mxs):
                mxs = cuurent 
            if(cuurent <0):
                cuurent =0
        return  mxs

File: 0053-maximum-subarray/0053-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) 
    {
        int cuurent = 0;
        int mxs =nums[0];
        for(auto i:nums)
        {
            cuurent+=i;
            if(cuurent>mxs)
            {
                mxs = cuurent;
            }
            if(cuurent <0)
            {
                cuurent =0;
            }
        }
        return  mxs;
        
    }
};